src/Tizen.BatteryMonitor/Tizen.BatteryMonitor/BatteryMonitor.cs

```diff
@@ -0,0 +1,159 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Tizen.BatteryMonitor
+{
+    static internal class Globals
+    {
+        internal const string LogTag = "Tizen.BatteryMonitor";
+    }
+
+    /// <summary>
+    /// Provides information related to the power consumption by applications or by hardware resources on a battery-powered device for a given duration of time.
+    /// </summary>
+    /// <feature>http://tizen.org/feature/battery</feature>
+    /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+    /// <since_tizen> 7 </since_tizen>
+    public static class BatteryMonitor
+    {
+        static BatteryMonitor()
+        {
+        }
+
+        private static double GetPowerUsageForResource(IntPtr BMHandle, ResourceType rtype)
+        {
+            double batteryUsage = 0;
+            int ret = Interop.BatteryMonitor.GetPowerUsageForResource(BMHandle, rtype, out batteryUsage);
+            if (((BatteryMonitorError)ret != BatteryMonitorError.None))
+            {
+                Log.Error(Globals.LogTag, "Error getting battery usage for resource" + (BatteryMonitorError)ret);
+                throw BatteryMonitorErrorFactory.ThrowBatteryMonitorException(ret);
+            }
+            return batteryUsage;
+        }
+
+        private static long ConvertDateTimeToTimestamp(DateTime dateTime)
+        {
+            DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
+            return (long)((dateTime.ToUniversalTime() - epoch).TotalSeconds);
+        }
+
+        /// <summary>
+        /// Gets the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.
+        /// </summary>
+        /// <since_tizen> 7 </since_tizen>
+        /// <param name="appID">Application ID of the application for which battery usage is required.</param>
+        /// <param name="start">Start Time for data in DateTime.</param>
+        /// <param name="end">End Time for data in DateTime.</param>
+        /// <param name="rtypes">list of resource type identifiers like BLE, WiFi, CPU etc.</param>
+        /// <returns>Returns the battery consumption in mAh(milli-Ampere hour) for the resources specified by the application in custom interval.</returns>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        public static IList<double> GetPowerUsageByAppForAllResources(string appID, DateTime start, DateTime end, IList<ResourceType> rtypes)
```

Comment:
How about overloading method? 
change parameter order to provide a consistency
```suggestion
        public static IList<double> GetPowerUsage(string appID, IList<ResourceType> rtypes, DateTime start, DateTime end)
```